internal/provider/resource_silo.go

```diff
@@ -0,0 +1,480 @@
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"description": schema.StringAttribute{
+							Description: "Certificate description.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"key": schema.StringAttribute{
+							Description: "PEM-formatted string containing private key.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"name": schema.StringAttribute{
+							Description: "Name associated to the certificate.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+							Validators: []validator.String{
+								stringvalidator.RegexMatches(regexp.MustCompile(`^[a-zA-Z0-9-]+$`),
+									"allowed characters are lowercase ASCII, digits, and \"-\""),
+							},
+						},
+						"service": schema.StringAttribute{
+							Description: "Service using this certificate.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+					},
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+			},
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Create: true,
+				Read:   true,
+				Update: true,
+				Delete: true,
+			}),
+			"time_created": schema.StringAttribute{
+				Computed:    true,
+				Description: "Timestamp of when this silo was created.",
+			},
+			"time_modified": schema.StringAttribute{
+				Computed:    true,
+				Description: "Timestamp of when this silo was last modified.",
+			},
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Unique, immutable, system-controlled identifier of the silo.",
+			},
+			"identity_mode": schema.StringAttribute{
+				Description: "How users and groups are managed in the Silo",
+				Computed:    true,
+				Default:     stringdefault.StaticString(string(oxide.SiloIdentityModeLocalOnly)),
+				Validators: []validator.String{
+					stringvalidator.OneOf(
+						string(oxide.SiloIdentityModeLocalOnly),
+						string(oxide.SiloIdentityModeSamlJit),
+					),
+				},
+			},
+		},
+	}
+}
+
+func siloCreateMappedFleetRolesModel(mappedFleetRoles map[string][]string) map[string][]oxide.FleetRole {
+	model := make(map[string][]oxide.FleetRole)
+
+	for key, fleetRoleModels := range mappedFleetRoles {
+		var roles []oxide.FleetRole
+		for _, frm := range fleetRoleModels {
+			roles = append(roles, oxide.FleetRole(frm))
+		}
+		model[key] = roles
+	}
+	return model
+}
+
+func newQuotasModel(quotas quotaResourceModel) oxide.SiloQuotasCreate {
+	return oxide.SiloQuotasCreate{
+		Cpus:    int(quotas.Cpus.ValueInt64()),
+		Memory:  oxide.ByteCount(quotas.Memory.ValueInt64()),
+		Storage: oxide.ByteCount(quotas.Storage.ValueInt64()),
+	}
+}
+
```

Comment:
This is only used in one place to save on an inline type declaration. We can just use the inline type declaration instead.

```suggestion
```